src/shared.js

```diff
@@ -792,9 +826,343 @@ export const getAllBoxes = async (flagValue) => {
+ * @param {object} specimen - specimen object from the Firestore.
+ */
+const removeUnusableTubes = (specimen) => {
+    const tubeDataObject = {};
+    tubeLoop: for (const tubeKey of specimenCollection.tubeCidList) {
+        const tube = specimen[tubeKey];
+
+        if (!tube || !tube[conceptIds.collection.tube.scannedId]) {
+            delete specimen[tubeKey];
+            continue;
+        }
+
+        if (tube[conceptIds.discardFlag] === conceptIds.yes) {
+            delete specimen[tubeKey];
+            continue;
+        }
+
+        if (tube[conceptIds.collection.tube.isMissing] === conceptIds.yes) {
+            delete specimen[tubeKey];
+            continue;
+        }
+
+        // This is a sanity check, but hasn't been needed in testing. All applicable tubes have been filtered by the discard flag.
+        const tubeDeviation = tube[conceptIds.collection.tube.deviation];
+        for (const deviationFlag of tubeDeviationFlags) {
+            if (tubeDeviation?.[deviationFlag] === conceptIds.yes) {
+                delete specimen[tubeKey];
+                continue tubeLoop;
+            }
+        }
+        tubeDataObject[tubeKey] = tube[conceptIds.collection.tube.scannedId];  
+    };
+    return tubeDataObject;
+}
+
+/**
+ * Get specimens from an array of collectionIds.
+ * @param {array<string>} collectionIdsArray - list of collectionIds to fetch specimen documents.
+ * @param {*} isBPTL - boolean to indicate if the request is from BPTL.
+ * @returns list of specimens objects.
+ */
+export const getSpecimensByCollectionIds = async (collectionIdsArray, isBPTL = false) => {
+  
+  if (collectionIdsArray.length === 0) return [];
+  const collectionIdQueryString = Array.from(collectionIdsArray).join(',');
+
+  try {
+      const idToken = await getIdToken();  
+      let queryString = `${api}api=getSpecimensByCollectionIds&collectionIdsArray=${collectionIdQueryString}`;
+      if (isBPTL) queryString += `&isBPTL=${isBPTL}`;
+      
+      const response = await fetch(queryString, {
+          method: 'GET',
+          headers: {
+          Authorization: 'Bearer ' + idToken,
+          }
+      });
+
+      if (!response.ok) throw new Error(response.statusText);
+
+      const specimensResponse = await response.json();
+      return specimensResponse.data;
+  } catch (error) {
+      console.error(error);
+      throw error;
+  }
+};
+
+/**
+ * Get specimens from a list of boxes.
+ * @param {array<object>} boxList - list of current unshipped boxes for the shipping dashboard.
+ * @param {boolean} isBPTL - boolean to indicate if the request is from BPTL.
+ * @returns list of specimens objects with only the specimens in the current boxes.
+ */
+export const getSpecimensInBoxes = async (boxList, isBPTL = false) => {
+  const { tubeIdSet, collectionIdSet } = extractCollectionIdsFromBoxes(boxList);
+  if (collectionIdSet.size === 0) return [];
+  const collectionIdQueryString = Array.from(collectionIdSet).join(',');
+
+  try {
```

Comment:
I like this idea - held off because I'm pretty sure it results in more code and `if/else` checks/handlers. I've asked Warren a similar question in the past and he opts for fewer lines, so I went with that.

I also thought about combining `getSpecimensByCollectionIds` and `getSpecimensInBoxes` but held off on that because I think the names are helpful/descriptive in the sections of code where they're called. Plus, it would add conditional complexity to the functions. I hope this thought process makes sense.